library/bignum.c

```diff
@@ -54,6 +54,131 @@
 #define MPI_VALIDATE(cond)                                           \
     MBEDTLS_INTERNAL_VALIDATE(cond)
 
+/*
+ * Compare signed values in constant time
+ */
+int mbedtls_mpi_lt_mpi_ct(const mbedtls_mpi *X,
+                          const mbedtls_mpi *Y,
+                          unsigned *ret)
+{
+    mbedtls_ct_condition_t cond, X_is_negative, Y_is_negative, result;
+
+    MPI_VALIDATE_RET(X != NULL);
+    MPI_VALIDATE_RET(Y != NULL);
+    MPI_VALIDATE_RET(ret != NULL);
```

Comment:
Since we removed most `NULL`-pointer checks on input parameters a while ago (insert link here), and since `MPI_VALIDATE_RET()` boils down to

```
#define MBEDTLS_INTERNAL_VALIDATE_RET(cond, ret)  do { } while (0)
```

is it worth adding such redundant checks in new code? People might think they are doing something!

Reply:
It's pre-existing (this code was moved across from `constant_time.c`). Agree we should remove use of this macro everywhere - but that's a separate clean-up, not in scope for this PR I think.